src/mon/Paxos.cc

```diff
@@ -1230,6 +1230,8 @@ void Paxos::trim()
 
   if (first_committed >= end)
     return;
+  if (get_version() > g_conf->paxos_max)
+    end = std::max(end, get_version() - g_conf->paxos_max);
```

Comment:
why do you think this helps? in your case, you might want to check why trim() is not called.

Reply:
i am not sure yet，and will check more log to verify。 now just want to take more discuss

Reply:
probably you could drop a mail to ceph-devel instead. i don't think a random PR would help to inspire the discussion.